[FLINK-24899][runtime] Enable data compression for blocking shuffle by default

SteNicholas opened the pull request:
## What is the purpose of the change

*Shuffle data compression is currently not enabled by default. Shuffle data compression is important for blocking data shuffle and enabling shuffle data compression by default can improve the usability. `taskmanager.network.blocking-shuffle.compression.enabled` should be true by default.*

## Brief change log

  - *Updates the default value of `taskmanager.network.blocking-shuffle.compression.enabled` to true.*

## Verifying this change

  - *The `testSortMergeShuffleConfigOptionsCorrelation` of `NettyShuffleEnvironmentConfigurationTest` adds the test to verify whether the defaule value of `taskmanager.network.blocking-shuffle.compression.enabled` is true.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

SteNicholas commented:
@wsry @zhuzhurk , could you please take a look at this pull request?

wsry commented:
Thanks for preparing this PR, I will take a look soon.

hililiwei commented on flink-core/src/main/java/org/apache/flink/configuration/NettyShuffleEnvironmentOptions.java:
```diff
@@ -76,7 +76,7 @@
     public static final ConfigOption<Boolean> BLOCKING_SHUFFLE_COMPRESSION_ENABLED =
             key("taskmanager.network.blocking-shuffle.compression.enabled")
-                    .defaultValue(false)
+                    .defaultValue(true)
```
```suggestion
                    .booleanType()
                    .defaultValue(true)
```
add type definition?

hililiwei reviewed (commented):
left some comments.  :-)

SteNicholas commented:
@hililiwei, thanks for your detailed review and I have addressed the above comment. IMO, the boolean type option has no the definition in `NettyShuffleEnvironmentOptions` and thus I have not added this definition.
@wsry , any comments?

SteNicholas commented:
@hililiwei, thanks for your detailed review and I have addressed the above comment. IMO, the boolean type option has no the definition in NettyShuffleEnvironmentOptions and thus I have not added this definition.
@wsry , could you please take a look at the changes?

wsry commented:
Hi, thanks a lot for the PR and review. I will take look after finishing this discussion: https://lists.apache.org/thread/pt2b1f17x2l5rlvggwxs6m265lo4ly7p.

wsry commented on flink-tests/src/test/java/org/apache/flink/test/runtime/ShuffleCompressionITCase.java:
```diff
@@ -84,8 +84,6 @@
     @Test
     public void testDataCompressionForBoundedBlockingShuffle() throws Exception {
```
After changing the default value to true, I guess many test cases including BlockingShuffleITCase have already covered the scenario of data compression. Here what we really need is a test case which covers the scenario where data is not compressed. Could you please change the method name including the below one to something like testBoundedBlockingShuffleWithoutCompression/testSortMergeBlockingShuffleWithoutCompression, and set the bool flag of data compression to false?

wsry commented:
@SteNicholas Thanks for your contribution, I have left a small comment.
BTW, could you please also update the corresponding document for blocking shuffle? (docs/content/docs/ops/batch/blocking_shuffle.md & docs/content.zh/docs/ops/batch/blocking_shuffle.md)

wsry commented:
The description for taskmanager.network.sort-shuffle.min-parallelism need also to be updated.

wsry commented:
@SteNicholas Thanks for the update, changes LGTM. I will help to merge it after the CI passes.

SteNicholas commented:
@wsry, I have added a commit for the failure of the CI. Please help to take a look at the update.

SteNicholas commented:
@flinkbot run azure
